front/src/modules/rollingStock/components/rollingStockEditor/RollingStockEditorCurves.tsx

```diff
@@ -197,44 +244,86 @@ export default function RollingStockEditorCurves({
+      /* * * 1st find the index of the modified curve in the curves list, * *
+       * * then slice the list to replace the selected curve by the new one */
+      const index = currentRsEffortCurve.modes[selectedTractionMode].curves.findIndex(
+        (curve) =>
+          curve.cond?.comfort === selectedComfortLvl &&
+          curve.cond?.electrical_profile_level === selectedElectricalProfile &&
+          curve.cond?.power_restriction_code === selectedPowerRestriction
+      );
+
+      const updatedSelectedCurve: ConditionalEffortCurve = {
+        cond: currentRsEffortCurve.modes[selectedTractionMode].curves[index].cond,
+        curve: parsedCurve,
+      };
 
-      setSelectedCurve((prevState) => ({
+      const updatedCurvesList = currentRsEffortCurve.modes[selectedTractionMode].curves
+        .slice(0, index)
+        .concat(
+          updatedSelectedCurve,
+          currentRsEffortCurve.modes[selectedTractionMode].curves.slice(index + 1)
+        );
+
+      const updatedCurrentRsEffortCurve = {
+        default_mode: currentRsEffortCurve.default_mode,
+        modes: {
+          ...currentRsEffortCurve.modes,
+          [selectedTractionMode]: {
+            ...currentRsEffortCurve.modes[selectedTractionMode],
+            curves: updatedCurvesList,
+          },
+        },
+      };
+
+      setCurrentRsEffortCurve(updatedCurrentRsEffortCurve);
+
+      // TODO: try to do it by only updating "currentRsEffortCurve" state.
+      let updatedSelectedCurvesList = updatedCurrentRsEffortCurve.modes[
+        selectedTractionMode
+      ]?.curves.filter((curve) => curve.cond?.comfort === selectedComfortLvl);
+
+      if (rollingstockParams?.powerRestrictions.length > 1)
+        updatedSelectedCurvesList = updatedSelectedCurvesList.filter(
+          (curve) => curve.cond?.electrical_profile_level === selectedElectricalProfile
+        );
+
+      setSelectedCurves({
         [selectedTractionMode]: {
-          ...prevState[selectedTractionMode],
-          curves: [
-            {
-              cond: selectedCurve[selectedTractionMode].curves[0].cond,
-              curve: parsedCurve,
-            },
-          ],
+          ...currentRsEffortCurve.modes[selectedTractionMode],
+          curves: updatedSelectedCurvesList,
         },
-      }));
+      });
     } else if (data) {
       setCurrentRsEffortCurve(data.effort_curves);
     }
   };
 
   useEffect(() => {
     updateRollingstockParams();
-  }, [selectedTractionMode, selectedComfortLvl]);
+  }, [selectedTractionMode, selectedComfortLvl, selectedElectricalProfile]);
 
   useEffect(() => {
-    const newSelectedCurve = selectCurve();
-    setSelectedCurve(newSelectedCurve);
-    setCurves(defaultCurve(newSelectedCurve));
-  }, [selectedComfortLvl, selectedTractionMode, selectedElectricalProfile]);
+    const newSelectedCurves = selectCurves();
+    setSelectedCurves(newSelectedCurves);
+    setCurves(defaultCurve(newSelectedCurves));
```

Comment:
This can be moved directly in selectCurves (which can be rename selectMatchingCurves, and in the description, you can add that they will be displayed in the graph RollingStockCurves and that the curve displayed in the spreadsheet will also be updated)
```suggestion
```